crates/sui-json-rpc/src/read_api.rs

```diff
@@ -816,6 +817,38 @@ impl ReadApiServer for ReadApi {
     async fn get_checkpoint(&self, id: CheckpointId) -> RpcResult<Checkpoint> {
         Ok(self.get_checkpoint_internal(id)?)
     }
+
+    async fn get_checkpoints(
+        &self,
+        cursor: Option<CheckpointSequenceNumber>,
+        limit: Option<usize>,
+        descending_order: bool,
+    ) -> RpcResult<CheckpointPage> {
+        let mut limit = limit.unwrap_or_default();
+        if limit > QUERY_MAX_RESULT_LIMIT_CHECKPOINTS || limit == 0 {
+            limit = QUERY_MAX_RESULT_LIMIT_CHECKPOINTS;
+        }
+
+        let mut data = self
+            .state
+            .get_checkpoints(cursor, Some(limit + 1), descending_order)?;
+
+        let has_next_page = data.len() > limit;
+
+        data.truncate(limit);
+
+        let mut next_cursor = Some(data.last().unwrap().sequence_number);
```

Comment:
Avoid using unwrap() in non test files unless you are 100% confident that this is not going to panic. In this case it might panic if data is empty